evalml/tests/automl_tests/test_iterative_algorithm.py

```diff
@@ -98,19 +97,21 @@ def test_iterative_algorithm_init(
     assert algo.batch_number == 0
     assert algo.default_max_batches == 1
     estimators = get_estimators(problem_type)
-    decomposer = [STLDecomposer] if is_regression(problem_type) else []
     assert len(algo.allowed_pipelines) == len(
         [
             make_pipeline(
                 X,
                 y,
                 estimator,
                 problem_type,
+                include_decomposer=include_decomposer,
                 parameters=search_parameters,
             )
             for estimator in estimators
-        ]
-        + decomposer,
+            for include_decomposer in (
+                [True, False] if is_regression(problem_type) else [False]
+            )
+        ],
```

Comment:
Sorry, I should have clarified with my request for parametrization - I meant that we should parametrize it over the whole function, like
```
@pytest.mark.parametrize("problem_type", ["binary", "multiseries time series regression"])
@pytest.mark.parametrize("include_decomposer", [True, False])
def test_iterative_algorithm_init(
    problem_type,
    include_decomposer
    X_y_binary,
    multiseries_ts_data_stacked,
):
if include_decomposer and problem_type=="binary":
    pytest.skip("Decomposers are not used in binary problems")
...
```
It makes it much easier to debug this way!

Reply:
To clarify, here's the intended behavior. 

- Binary problems have 6 estimators: LightGBM, Extra Trees, Elastic Net, XGBoost, RF, and Logistic Regression. Decomposers cannot be used with binary problems. As such, we want six pipelines. 
   - LightGBM Pipeline
   - Extra Trees Pipeline
   - Elastic Net Pipeline
   - XGBoost Pipeline
   - RF Pipeline
   - Logistic Regression Pipelines 

For a total of 5 pipelines. When we evaluate what we generate with `len(algo.allowed_pipelines)`, this evaluates to true. 

- For multiseries time series problems, there are 5 estimators: VARMAX, Extra Trees, XGBoost, RF, and Elastic Net. Since decomposers can be used with multiseries problems, we want to generate one pipeline with a decomposer and one pipeline without a decomposer. 
   - VARMAX w/ Decomposer Pipeline
   - VARMAX w/o Decomposer Pipeline 
   - Extra Trees w/ Decomposer Pipeline
   - Extra Trees w/o Decomposer Pipeline 
   - XGBoost w/ Decomposer Pipeline
   - XGBoost w/o Decomposer Pipeline 
   - RF w/ Decomposer Pipeline
   - RF w/o Decomposer Pipeline 
   - Elastic Net w/ Decomposer Pipeline
   - Elastic Net w/o Decomposer Pipeline 

For a total of 10 pipelines. When we evaluate what we generate with `len(algo.allowed_pipelines)`, this evaluates to true. 

Thus, I don't think making `include_decomposer` a test parameter makes sense since we're using this variable to make the pipelines with and without the decomposer when checking for the `mutliseries time series regression` problem type. We only need to test two cases here, which is (binary pipelines without decomposer) and (multiseries pipelines with and without decomposers).

As always, open to including clarifying comments or a cleaner implementation.

Reply:
Ooh I see. Makes sense! Sorry to drag you down this false path with me haha